msal/src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -156,21 +158,30 @@ internal void OnBeforeWrite(TokenCacheNotificationArgs args)
             if (idToken != null)
             {
                 msalIdTokenCacheItem = new MsalIdTokenCacheItem
-                    (preferredEnvironmentHost, requestParams.ClientId, response, tenantId);
+                    (preferredEnvironmentHost, requestParams.ClientId, response, tenantId, subject);
             }
 
             lock (LockObject)
             {
                 try
                 {
+                    Account account = null;
+                    if(msalAccessTokenCacheItem.HomeAccountId != null)
+                    {
+                        if(requestParams.Authority.AuthorityType == Core.Instance.AuthorityType.Adfs)

```

Comment:
>if [](start = 24, length = 2)

I know that this isn't your code, but could you: 

a. Move this block to a private method (this one is too large)
b. Use a switch (Authority) here instead of if / else. Similar to the switch in Authority.cs you've added.

Reply:
Condensed code, but didn't add new function